### PR TITLE
Run tests on Node 20 (requires pnpm 8)

### PR DESCRIPTION
**Pull request by thgh**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,6 +23,8 @@ jobs:
           fetch-depth: 0
 
       - uses: pnpm/action-setup@v2.2.4
+        with:
+          version: 8
```
Comment (huozhi) [inline .github/workflows/ci.yml:27]: pnpm 8 is different from the configured version in package.json, need to update there with a specific (8.x.x) version and lock file as well.

**Review by huozhi — COMMENTED** [review]
CI error: 

```
Run pnpm install --frozen-lockfile --prefer-offline
 ERR_PNPM_LOCKFILE_CONFIG_MISMATCH  Cannot proceed with the frozen installation. The current "settings.autoInstallPeers" configuration doesn't match the value found in the lockfile
 ```

**Comment by thgh** [comment]
I have not used pnpm before so I'm not sure if `autoInstallPeers: true` will cause issues.

**Comment by huozhi** [comment]
Taking a look 👀

**Comment by huozhi** [comment]
Did a force re-install locally looks like it's marked as `autoInstallPeers: true` by default with pnpm. Looks good! Thanks!

**Pull request merged**